### PR TITLE
Add PHP 8.4 `array_*` polyfill

### PR DESCRIPTION
**Pull request by joshuaruesweg**
See: https://wiki.php.net/rfc/array_find

### CODE REVIEW

File: src/Php84/Php84.php [hunk 1]
```diff
@@ -63,4 +63,48 @@ public static function mb_lcfirst(string $string, ?string $encoding = null): str
 
         return $firstChar . mb_substr($string, 1, null, $encoding);
     }
+
+    public static function array_find(array $array, callable $callback)
+    {
+        foreach ($array as $key => $value) {
+            if ($callback($value, $key)) {
```
Comment (keradus) [inline src/Php84/Php84.php:70]:
```suggestion
            if (true === $callback($value, $key)) {
```

i really wonder why not to rely on strict, boolean interface of callback result

Comment (joshuaruesweg) [reply src/Php84/Php84.php:70]: Thus typecheck would not match the nativ implementation which also has no strict typecheck: 

```
var_dump(array_find(['', 'aa'], fn($value) => strlen($value))); // string(2) "aa"
```

**Comment by Ayesh** [comment]
LGTM, you are the RFC author after all :)

I also wrote a polyfill as a separate package, and used a [conditional declaration](https://github.com/PHP-Polyfills/array-find/blob/69874552b2c2c988585929d722e292a3a859e673/src/array_find.php#L52) to add the `mixed` return type only in PHP >=8.0

**Review by IonBazan — APPROVED** [review]

File: tests/Php84/Php84Test.php [hunk 2]
```diff
@@ -30,6 +30,34 @@ public function testMbLcFirst(string $string, string $expected): void {
         $this->assertSame($expected, mb_lcfirst($string));
     }
 
+    /**
+     * @dataProvider arrayFindDataProvider
+     */
+    public function testArrayFind(array $array, callable $callback, $expected): void {
```
Comment (OskarStark) [inline tests/Php84/Php84Test.php:36]: Don't we ensure our Symfony CS here in this repo?

Comment (nicolas-grekas) [reply tests/Php84/Php84Test.php:36]: not automatically, I run php-cs-fixer from time to time for the moment

Comment (OskarStark) [reply tests/Php84/Php84Test.php:36]: OK

**Review by vudaltsov — APPROVED** [review]

**Comment by nicolas-grekas** [comment]
Thank you @joshuaruesweg.

**Pull request merged**